src/fs/filesystem.ts

```diff
@@ -138,27 +162,44 @@ export class FileSystem {
 
   async runOnTree<a>(
     path: string,
-    updateTree: boolean,
+    updateTree: boolean, // ie. do a mutation
     fn: (tree: Tree, relPath: string) => Promise<a>
   ): Promise<a> {
     const parts = pathUtil.split(path)
     const head = parts[0]
     const relPath = pathUtil.join(parts.slice(1))
+
     let result: a
+    let resultPretty: a
+
     if (head === 'public') {
       result = await fn(this.publicTree, relPath)
+
       if (updateTree && PublicTree.instanceOf(result)) {
+        resultPretty = await fn(this.prettyTree, relPath)
+
         this.publicTree = result
+        this.prettyTree = resultPretty as unknown as PublicTree
       }
+
     } else if (head === 'private') {
```

Comment:
This looks like a job for OO classes

Reply:
Yeah possibly, not sure I'd fix this here though. This `head` part is part of the path the user provides, for example, `fs.ls("private/dir")`. We might need to revisit this in another PR and do something like `fs.ls("dir", { mode: fs.modes.private })`.